### ISSUE TITLE
Nvidia "glXChooseVisual failed" error

### ISSUE THREAD

**Issue by GloriousEggroll** [solus-project/linux-steam-integration#44]
Just tried to run the snap from my other box which has an nvidia 1050 in it. After I agreed to the TOS on steam, it updated, then gave me a dialog with glXChooseVisual failed. I noticed you had a full-nvidia-support branch of snapd, then checked the official one from snapcore, which has the same changes. System is Arch linux + latest proprietary drivers + gtx 1050

**Comment by ikeydoherty** [comment]
You need the 32-bit drivers installed too, you got those?

**Comment by GloriousEggroll** [comment]
should be, nvidia-libgl and lib32-nvidia-libgl are part of nvidia-utils and lib32-nvidia-utils. see attached:
![screenshot_20171218_144540](https://user-images.githubusercontent.com/11287837/34124791-3689fd7c-e402-11e7-9332-cc6b1f84d92d.png)

**Comment by ikeydoherty** [comment]
Were they installed prior to running the new snapd version? If not you need a reboot.

**Comment by GloriousEggroll** [comment]
yes, ive had them installed on this machine since i installed the os. ( i dont use install scripts for arch)

**Comment by ikeydoherty** [comment]
Can you check if you're able to run both `linux-steam-integration.glxgears` and `linux-steam-integration.glxgears-32` ?

Also please ensure you're only using the snaps from the store, not locally installed versions.

Both in the edge channel, `linux-steam-integration` requires `--devmode`

**Comment by GloriousEggroll** [comment]
alright, will try that momentarily. no glxgears or glxgears-32 in the packages I used. I used these:
wget https://packages.solus-project.com/lsi/solus-runtime-gaming_0.0.0_amd64.snap
wget https://packages.solus-project.com/lsi/linux-steam-integration_0.6_amd64.snap

[thomas@TOM-PC snapd-git]$ snap run linux-steam-integration.glxgears
error: cannot find app "glxgears" in "linux-steam-integration"

**Comment by ikeydoherty** [comment]
Oh god yea those are ancient, ok remove both of those, then:

```
sudo snap install --edge solus-runtime-gaming
sudo snap install --devmode --edge linux-steam-integration
```

Because of the way snapd uses tmpfs namespaces you'll almost certainly need a reboot now

**Comment by ikeydoherty** [comment]
Once we get past this I'll look at getting a new revision together too so we can get the unity-black-screen-of-nope fix out

**Comment by GloriousEggroll** [comment]
[thomas@TOM-PC quorrabot]$ snap run linux-steam-integration.glxgears
*** stack smashing detected ***: /snap/linux-steam-integration/10/bin/glxgears terminated
Segmentation fault (core dumped)

[thomas@TOM-PC quorrabot]$ snap run linux-steam-integration
Setting up Steam content in /home/thomas/snap/linux-steam-integration/common/.local/share/Steam
Running Steam on Linux 4.14.6-1-ARCH 64-bit
STEAM_RUNTIME is disabled by the user
Installing breakpad exception handler for appid(steam)/version(0)
/home/thomas/snap/linux-steam-integration/common/.local/share/Steam/steam.sh: line 713:  1345 Segmentation fault      (core dumped) $STEAM_DEBUGGER "$STEAMROOT/$STEAMEXEPATH" "$@"
/home/thomas/snap/linux-steam-integration/common/.local/share/Steam/steam.sh: line 444: no match: ssfn*

[thomas@TOM-PC quorrabot]$ snap run linux-steam-integration.glxgears-32
Segmentation fault (core dumped)

**Comment by ikeydoherty** [comment]
What version of snapd is this..?

**Comment by GloriousEggroll** [comment]
i'm running snapd-git

**Comment by AccountOneOff** [comment]
Installed LSI today and I'm getting the same glXChooseVisual failed error on a GTX 960.

    $ snap run linux-steam-integration
    Running Steam on ubuntu 17.10 64-bit
    STEAM_RUNTIME is disabled by the user
    Installing breakpad exception handler for appid(steam)/version(1513371133)
    libGL error: No matching fbConfigs or visuals found
    libGL error: failed to load driver: swrast
    Installing breakpad exception handler for appid(steam)/version(1513371133)
    Installing breakpad exception handler for appid(steam)/version(1513371133)
    Gtk-Message: Failed to load module "unity-gtk-module"
    Gtk-Message: Failed to load module "canberra-gtk-module"
    Installing breakpad exception handler for appid(steam)/version(1513371133)
    libGL error: No matching fbConfigs or visuals found
    libGL error: failed to load driver: swrast
    glXChooseVisual failedMain.cpp (322) : Assertion Failed: Fatal Error: glXChooseVisual failed
    Assert( Assertion Failed: Fatal Error: glXChooseVisual failed ):Main.cpp:322

    Installing breakpad exception handler for appid(steam)/version(1513371133)
    No minidump written, nothing to upload.

    (process:3769): GConf-CRITICAL **: gconf_client_notify_remove: assertion 'client->listeners != NULL' failed

    (process:3769): GConf-CRITICAL **: gconf_client_notify_remove: assertion 'client->listeners != NULL' failed

I was able to run `linux-steam-integration.glxgears` but not `linux-steam-integration.glxgears-32`

    $ snap run linux-steam-integration.glxgears-32
    libGL error: No matching fbConfigs or visuals found
    libGL error: failed to load driver: swrast
    Error: couldn't get an RGB, Double-buffered visual

- snapd/artful-updates,now 2.29.4.2+17.10 amd64 [installed]
- nvidia-387/artful,now 387.34-0ubuntu0~gpu17.10.2 amd64 [installed] via graphics-drivers/ppa
- Ubuntu 17.10

**Comment by AccountOneOff** [comment]
Updated to Nvidia 390.12 and I no longer receive the glx error.

**Comment by Toasterson** [comment]
Be sure to have the 32bit version of the nvidia drivers installed aswell. on Arch it's in different package

**Comment by milipili** [comment]
Same problem with debian testing since a few weeks:
```bash
$ sudo apt install nvidia-driver:i386                                                                                                                                                                                                                                                                                                                                                                                          [...]

The following packages have unmet dependencies:
 nvidia-driver:i386 : Depends: xserver-xorg-video-nvidia:i386 (= 384.111-4) but it is not going to be installed
E: Unable to correct problems, you have held broken packages.
```

Actually I don't understand why we need the 32bits version. It does not make so much sense in a 64bits version of the system. Why is it that difficult to produce 2 binaries for steam?

**Comment by z0al** [comment]
@milipili I'm on debian 9 and same issue. Did you manage to solve it?

**Comment by milipili** [comment]
This has been fixed by the upstream packages. that said you may have to force or to install manually some packages.
For info there is a current regression in performance in debian testing, everything related to opengl has became awfully slow

**Comment by milipili** [comment]
Actually i think that should be closed as the problem has been solved already
